client/dashboard/top-selling-products/index.js

```diff
@@ -81,21 +81,37 @@ export class TopSellingProducts extends Component {
 		} );
 	}
 
-	render() {
+	getCardContents( title ) {
 		const { data, isRequesting, isError } = this.props;
 
-		// @TODO We will need to update it with a error/empty data indicator
 		const rows = isRequesting || isError ? [] : this.getRowsContent( data );
 		const headers = this.getHeadersContent();
+
+		if ( isRequesting ) {
+			return <TablePlaceholder caption={ title } headers={ headers } />;
+		}
+
+		if ( isError ) {
+			// @TODO An error notice should be displayed when there is an error
+		}
+
+		if ( rows.length === 0 ) {
+			return (
+				<div className="woocommerce-top-selling-products__empty-message">
```

Comment:
🤔 should we consider extracting this to a generic <DashboardCardMessage /> component? Since it is likely all other dashboard blocks/components will need to have a similar messaging state for no records / errors. I suppose we could leave it as-is for now and maybe extract it out when we build the next block for the dashboard.

Reply:
Sure, we can easily extract it. I wonder how that component should be named and where should it be in the codebase. Some ideas:
- `EmptyCard`: if it will be displayed in most empty cards.<sup>1</sup>
- `EmptyTable`: if it will be displayed instead of empty tables.
- `EmptyMessage`: if it will be displayed in different places other than cards and tables.

<sup>1</sup> We might even modify the `Card` component to render `EmptyCard` when no children is passed in the props.